pkgs/native_assets_builder/lib/src/build_runner/build_runner.dart

```diff
@@ -45,7 +43,7 @@ class NativeAssetsBuildRunner {
     int? targetAndroidNdkApi,
     required bool includeParentEnvironment,
   }) async {
-    _metadata.clear();
+    _clear();
```

Comment:
It would be cleaner not to have state that is modified as side-effects of calling functions, then clearing such state, ...

Seems cleaner to have `_buildPackageCached()` return a `(assets, errors, dependencies)` record